src/Tribe/Editor/Compatibility/Classic_Editor.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+namespace Tribe\Editor\Compatibility;
+/**
+ * Editor Compatibility with classic editor plugins.
+ *
+ * https://dev.tec/wp-admin/post.php?post=796&action=edit&classic-editor
+ * https://dev.tec/wp-admin/post.php?post=796&action=edit&classic-editor__forget
+ * https://dev.tec/wp-admin/post.php?post=796&action=edit&classic-editor&classic-editor__forget
+ *
+ * @since TBD
+ */
+class Classic_Editor {
+	/**
+	 * "Classic Editor" flag for blocks/classic
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $classic_option_key = 'classic-editor-replace';
+
+	/**
+	 * "Classic Editor" original param for blocks->classic.
+	 * Can be overridden by user choice.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $classic_param = 'classic-editor';
+
+	/**
+	 * "Classic Editor" term used for comparisons.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $classic_term = 'classic';
+
+	/**
+	 * "Blocks Editor" term used for comparisons.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $block_term = 'block';
+
+	/**
+	 * "Classic Editor" param for user override
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $classic_override = 'classic-editor__forget';
+
+	/**
+	 * "User Choice" key for user override
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $user_choice_key = 'classic-editor-allow-users';
+
+	/**
+	 * User meta "User Choice" key for user override
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $user_meta_choice_key = 'classic-editor-settings';
+
+	/**
+	 * Post meta key used for CE "remembering" the last editor used.
+	 * The bane of my existence.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $post_meta_key = 'classic-editor-remember';
+
+	/**
+	 * Stores the values used by the Classic Editor plugin to indicate we're using the classic editor.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	public static $classic_values  = [
+		'replace',
+		'classic',
+	];
+
+	/**
+	 * Placeholders
+	 *
+	 * @since TBD
+	 *
+	 * @var [type]
+	 */
+	public static $user_choice_allowed = null;
+	public static $user_profile_choice = null;
+	public static $classic_url_param = null;
+	public static $classic_url_override = null;
+
+	/**
+	 * Registers the hooks and filters required based on if the Classic Editor plugin is active.
+	 *
+	 * @since TBD
+	 */
+	public function init() {
+		if ( static::is_classic_plugin_active() ) {
+			$this->hooks();
+		}
+	}
+
+	/**
+	 * Hooks for loading logic outside this class.
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'set_classic_url_params' ], 22 );
+
+		global $current_screen;
+		$good_screens = [
```

Comment:
Should we add a check here if the current screen is empty?

Reply:
Actually, this needs to be removed, this function gets called a few times before `$current_screen` is available 🤦🏼